src/pages/dataElementGroups/fields/NameField.tsx

```diff
@@ -0,0 +1,64 @@
+import i18n from '@dhis2/d2-i18n'
+import { InputFieldFF } from '@dhis2/ui'
+import React, { useMemo } from 'react'
+import { Field as FieldRFF, useField } from 'react-final-form'
+import { useParams } from 'react-router-dom'
+import {
+    composeAsyncValidators,
+    required,
+    useCheckMaxLengthFromSchema,
+    useIsFieldValueUnique,
+} from '../../../lib'
+import { SchemaName } from '../../../types'
+import type { FormValues } from '../form'
+
+function useValidator() {
+    const params = useParams()
+    const dataElementId = params.id as string
+    const checkIsValueTaken = useIsFieldValueUnique({
+        model: 'dataElements',
```

Comment:
`dataElementGroups`?

Reply:
I think fields like `Name`, `description`, `ShortName`, `Code` could easily be extracted to a common component, and just pass what you need, or use the field context from `useSchema` and `useSection` (or `useSchemaSection`).

Reply:
Example of generic `NameField`:

```ts

function useValidator() {
    const params = useParams()
    const schemaSection = useSchemaSectionHandleOrThrow()

    const modelId = params.id as string
    const checkIsValueTaken = useIsFieldValueUnique({
        model: schemaSection.namePlural,
        field: 'name',
        id: modelId,
    })

    const checkMaxLength = useCheckMaxLengthFromSchema(
        schemaSection.name,
        'name'
    )

    return useMemo(
        () =>
            composeAsyncValidators<string, FormValues>([
                checkIsValueTaken,
                checkMaxLength,
                required,
            ]),
        [checkIsValueTaken, checkMaxLength]
    )
}

export function NameField({ helpText }: { helpText?: string }) {
    const validator = useValidator()
    const { meta } = useField('name', {
        subscription: { validating: true },
    })

    return (
        <FieldRFF<string | undefined>
            loading={meta.validating}
            component={InputFieldFF}
            dataTest="formfields-name"
            required
            inputWidth="400px"
            label={i18n.t('{{fieldLabel}} (required)', {
                fieldLabel: i18n.t('Name'),
            })}
            name="name"
            helpText={helpText}
            validate={(name?: string) => validator(name)}
            validateFields={[]}
        />
    )
}

```